src/panels/lovelace/cards/hui-history-graph-card.ts

```diff
@@ -105,23 +109,32 @@ export class HuiHistoryGraphCard extends LitElement implements LovelaceCard {
 
   protected updated(changedProps: PropertyValues) {
     super.updated(changedProps);
-    if (!this._config || !this.hass || !this._cacheConfig) {
+    if (
+      !this._config ||
+      !this.hass ||
+      !this._throttleGetStateHistory ||
+      !this._cacheConfig
+    ) {
       return;
     }
 
     if (!changedProps.has("_config") && !changedProps.has("hass")) {
       return;
     }
 
-    const oldConfig = changedProps.get("_config") as HistoryGraphCardConfig;
+    const oldConfig = changedProps.get("_config") as
+      | HistoryGraphCardConfig
+      | undefined;
 
-    if (changedProps.has("_config") && oldConfig !== this._config) {
-      this._getStateHistory();
-    } else if (
-      this._cacheConfig.refresh &&
-      Date.now() - this._date!.getTime() >= this._cacheConfig.refresh * 100
+    if (
+      changedProps.has("_config") &&
+      (oldConfig?.entities !== this._config.entities ||
+        oldConfig?.hours_to_show !== this._config.hours_to_show)
```

Comment:
```suggestion
      oldConfig &&
      (oldConfig.entities !== this._config.entities ||
        oldConfig.hours_to_show !== this._config.hours_to_show)
```

Reply:
`changedProps.has("_config")` !== `changedProps.get("_config")` it can be changed from undefined to a value, which would make `get` undefined but `has` true.